src/sage/graphs/graph_decompositions/tree_decomposition.pyx

```diff
@@ -768,12 +758,186 @@ def treewidth(g, k=None, kmin=None, certificate=False, algorithm=None):
     G.add_edges(((Set(x), Set(y)) for x, y in TD), loops=False)
 
     # The Tree-Decomposition may contain a lot of useless nodes.
-    # We merge all edges between two sets S,S' where S is a subset of S'
+    # We merge all edges between two sets S, S' where S is a subset of S'
     G = reduced_tree_decomposition(G)
 
     G.name("Tree decomposition")
     return G
 
+def make_nice_tree_decomposition(T):
+    r"""
+    Return a *nice* tree decomposition (TD) of the TD `T`.
+
+    A *nice* TD `NT` is a rooted tree with four types of nodes:
```

Comment:
don't forget to add a reference to a paper defining a nice tree decomposition.

The reference should be added to file `src/doc/en/reference/references/index.rst`.